packages/adaptivecpp/package.py

```diff
@@ -0,0 +1,252 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import json
+from os import path
+from glob import glob
+
+from llnl.util import filesystem
+
+from spack import *
+
+"""
+Install nvc++ version with soemthing like
+
+    spack install neso.adaptivecpp +nvcxx
+
+# openmp only
+    spack install neso.adaptivecpp
+to avoid llvm build
+"""
+
+
+class Adaptivecpp(CMakePackage):
+    """AdaptiveCPP is an implementation of the SYCL standard programming model
+    over NVIDIA CUDA/AMD HIP"""
+
+    homepage = "https://github.com/AdaptiveCpp/AdaptiveCpp"
+    git = "https://github.com/AdaptiveCpp/AdaptiveCpp.git"
+
+    provides("sycl")
+
+    version(
+        "24.02.0",
+        commit="974adc33ea5a35dd8b5be68c7a744b37482b8b64",
+        submodules=True,
+    )
+    version(
+        "23.10.0",
+        commit="3952b468c9da89edad9dff953cdcab0a3c3bf78c",
+        submodules=True,
+    )
+
+    variant(
+        "cuda",
+        default=False,
+        description="Enable CUDA backend for SYCL kernels using llvm+cuda",
+    )
+    variant(
+        "nvcxx",
+        default=False,
+        description="Enable CUDA backend for SYCL kernels using nvcxx",
+    )
+    variant(
+        "omp_llvm",
+        default=False,
+        description="Enable accelerated OMP backend for SYCL kernels using LLVM",
+    )
+    variant(
+        "opencl",
+        default=False,
+        description="Enable OpenCL backend.",
+    )
+
+    depends_on("cmake@3.5:", type="build")
+    depends_on("boost +filesystem", when="@23.10.0:")
+    depends_on(
+        "boost@1.60.0: +filesystem +fiber +context cxxstd=17", when="@23.10.0:"
+    )
+    depends_on("python@3:")
+    # depends_on("llvm@8: +clang", when="~cuda")
+    depends_on("llvm@9: +clang", when="+cuda")
+    depends_on("llvm@9: +clang", when="+omp_llvm")
+    depends_on("cuda", when="@23.10.0: +cuda")
+    depends_on("cuda", when="+nvcxx")
+    depends_on("nvhpc@22.9:", when="+nvcxx", type="build")
+    depends_on("opencl@3.0", when="+opencl")
```

Comment:
How does this work with different compilers? E.g., is it able to use GCC with `+nvcxx` or `+omp_llvm` variants? Or can it only use `nvcxx` and `clang` (respectively) in those cases?

Reply:
I've almost exclusively used +nvcxx with GCC. I find that the nvhpc packages does not need to be loaded downstream when you use acpp to compile NESO (with cuda-nvcxx backend). But if you do set nvhpc to be a run dependency then the nvhpc package gets loaded when acpp gets loaded. Loading nvhpc then sets CC,CXX to be the nvidia compiler globally which breaks other compilation paths, e.g. omp.library-only.

Reply:
I have not used omp_llvm for a while (again inherited from the hipsycl package) usually the host `omp.library-only` and `cuda-nvcxx` backends coexist within an single install without issue.

Reply:
OK, I just wasn't sure how it all worked.